tests/layering/node_cmd_test.go

```diff
@@ -0,0 +1,436 @@
+		Stdin:     runOpts.Stdin != nil,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}
+
+	req := restClient.Post().
+		Resource("pods").
+		Name(cmdPod.Name).
+		Namespace(cmdPod.Namespace).
+		SubResource("exec")
+
+	req.VersionedParams(execOpts, scheme.ParameterCodec)
+
+	// TODO: Figure out a better way to get the config from our clientset than
+	// having to read it back in.
+	kubeconfig, err := n.clientSet.GetKubeconfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not get kubeconfig: %w", err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not get config: %w", err)
+	}
+
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return nil, fmt.Errorf("could not get command executor: %w", err)
+	}
+
+	stdinBuf := bytes.NewBuffer([]byte{})
+	stdoutBuf, stdout := getWriterAndBuffer(runOpts.Stdout)
+	stderrBuf, stderr := getWriterAndBuffer(runOpts.Stderr)
+
+	streamOpts := remotecommand.StreamOptions{
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    false,
+	}
+
+	// Wire in stdin using io.TeeReader so its contents are available in the result object.
+	if runOpts.Stdin != nil {
+		streamOpts.Stdin = io.TeeReader(runOpts.Stdin, stdinBuf)
+	}
+
+	// Run the actual command
+	start := time.Now()
+	err = exec.Stream(streamOpts)
+	end := time.Since(start)
+
+	results := CmdResult{
+		Command:  runOpts.Command,
+		Duration: end,
+		Stdin:    stdinBuf.Bytes(),
+		Stdout:   stdoutBuf.Bytes(),
+		Stderr:   stderrBuf.Bytes(),
+	}
+
+	if err != nil {
+		err = &NodeCmdError{
+			CmdResult: results,
+			execErr:   err,
+		}
+	}
+
+	return &results, err
+}
+
+// Creates a new buffer and io.MultiWriter so we can collect stdin / stderr to
+// multiple places simultaneously.
+func getWriterAndBuffer(w io.Writer) (*bytes.Buffer, io.Writer) {
+	buf := bytes.NewBuffer([]byte{})
+	if w == nil {
+		return buf, buf
+	}
+
+	return buf, io.MultiWriter(buf, w)
+}
+
+// Prepends chroot /host onto the command we want to run.
+func getCommandToRun(cmd []string) []string {
+	if strings.HasPrefix(strings.Join(cmd, " "), "chroot /host") {
+		return cmd
+	}
+
+	return append([]string{"chroot", "/host"}, cmd...)
+}
+
+// Determines if the command pod is ready. These checks might be a little
+// heavy-handed, but they work well.
+func isPodReady(pod *corev1.Pod) bool {
+	// Check that the pod is not in the running phase
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// Check all the pod conditions
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodInitialized && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.PodScheduled && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.PodReady && condition.Status != "True" {
+			return false
+		}
+
+		if condition.Type == corev1.ContainersReady && condition.Status != "True" {
```

Comment:
q: if containers are ready and condition.status == true can the below ever be valid checks? (I think I'm asking: is containersReady a proxy for the check below??)

Reply:
Honestly, I wasn't sure what I needed to check and went a bit overboard. Just this check alone might be sufficient.